### PR TITLE
Table function support 

### PR DESCRIPTION
**Pull request by fulghum**
Add support for table_functions as a new type of `table factor` in the dolthub/vitess SQL grammar.

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -3903,6 +3903,7 @@ table_references:
 table_reference:
   table_factor
 | join_table
+| table_function
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:3906]: Probably better expressed as a type of table factor, so you can alias it

Comment (fulghum) [reply go/vt/sqlparser/sql.y]: Thanks; I went ahead and moved table_function to be a table_factor. There's more work needed to enable aliases for table functions, but I'm deprioritizing that in favor of getting the initial support for table functions out faster. Happy to return to enable aliases of course.

File: go/vt/sqlparser/parse_test.go [hunk 2]
```diff
@@ -39,6 +39,24 @@ type parseTest struct {
...
+		},
+		{
+			input:  "select * from my_table_function('foo')",
+			output: "select * from my_table_function('foo')",
+		},
+		{
+			input:  "select * from my_table_function('foo', 'bar')",
+			output: "select * from my_table_function('foo', 'bar')",
+		},
+		// TODO: We should be able to alias table functions, but that will require additional changes
```
Comment (zachmu) [inline go/vt/sqlparser/parse_test.go:54]: No commented out tests. Put them in their own test block and run them with t.Skip() instead

**Review by zachmu — APPROVED** [review]
LGTM, just one comment

File: go/vt/sqlparser/parse_test.go (hunk 2 above)
Comment (fulghum) [reply go/vt/sqlparser/parse_test.go]: Good rule. I went ahead and pulled this out. It'll be super easy to add back when we add alias support.

**Pull request merged**